add ability to unlink and link existing issues

macqueen opened the pull request:
![screenshot 2016-06-15 16 38 00](https://cloud.githubusercontent.com/assets/5026776/16101065/a528120c-3317-11e6-9ba5-a41cf764ab2f.png)
![screenshot 2016-06-15 16 40 27](https://cloud.githubusercontent.com/assets/5026776/16101134/33ab0728-3318-11e6-975b-f7d9282fabb9.png)

- also changed text on button from 'Create GitHub Issue' to 'Link GitHub Issue' so it's more generic
- also added linking existing issues to the activity stream (but not unlinking issues... not sure if we want that?)

@getsentry/api

macqueen commented on src/sentry_github/plugin.py:
```diff
@@ -97,6 +112,39 @@ def get_allowed_assignees(self, request, group):
+        issues = []
+        url = self.build_api_url(group, 'issues', query_params={'state': 'open', 'per_page': 100})
+        num_requests = 0
+        while url and num_requests < 10:
```
probably need to figure out a good way to cache this since this could be multiple requests per form init... suggestions?

macqueen commented:
ok @dcramer I updated this to use the select2 ajax option (and the github issue search api instead of issue list). I also added the ability to add an optional comment when linking to existing issues:

![screenshot 2016-06-21 18 34 21](https://cloud.githubusercontent.com/assets/5026776/16251966/47dbbbc2-37df-11e6-89c2-e2f10681c2fb.png)

dcramer commented:
:+1:

maybe @mattrobenolt can do a more thorough review, but general approach seems eric

macqueen commented:
moved issue number so it's first and also added a link in help text to GitHub's issue search syntax.

![screenshot 2016-06-22 13 45 15](https://cloud.githubusercontent.com/assets/5026776/16282971/bf401d02-387f-11e6-9b10-917661ee4a60.png)

mattrobenolt commented on src/sentry_github/plugin.py:
```diff
@@ -65,38 +84,58 @@ class GitHubPlugin(IssuePlugin):
 
         return (('', 'Unassigned'),) + users
 
+    def get_link_existing_issue_form(self, request, group, event, **kwargs):
```
Couldn't we move this into the parent `IssuePlugin` class? This looks like it'd be reusable for every plugin that wanted this functionality.

Then you could add a new property onto the class like, `link_issue_form = GitHubExistingIssueForm`.

macqueen replied:
sure i can do that. i hadn't because originally i was doing something more github specific in that method, but i can just change [this](https://github.com/getsentry/sentry/pull/3495/files#diff-0e1bf00279c4b111e1f181be05cf3474R123) to be your suggestion.

mattrobenolt replied:
Ah. Yeah, I'd do that, then add a method also for `get_initial_link_form_data`.

These are the same as mirroring the logic of `get_initial_form_data` and `get_new_issue_form`.

macqueen replied:
yup, sounds good!

aryeh-looker commented:
This is great! 👍👍👍
